lib/algorand.encoder.ts

```diff
@@ -110,4 +110,18 @@ export class AlgorandEncoder extends Encoder{
 		const decoded: object = msgpack.decode(encoded) as object
 		return decoded as object
 	}
+
+	/**
+	 * Casts a number or bigint to BigInt and checks if it's within the safe integer range.
+	 * @param value - The number or bigint to be casted.
+	 * @returns The value as a BigInt.
+	 * @throws Error if the value is not within the safe integer range.
+	 */
+	static safeCastBigInt(value: number | bigint): bigint {
+		const bigIntValue = BigInt(value)
+		if (bigIntValue < Number.MIN_SAFE_INTEGER || bigIntValue > Number.MAX_SAFE_INTEGER) {
```

Comment:
This just needs a type check the passed in value to make sure it is a number first before throwing an Error

```suggestion
if (typeof value === "number" && (value < Number.MIN_SAFE_INTEGER || value > Number.MAX_SAFE_INTEGER)) {
```